tests/lib/rules/no-unneeded-ternary.js

```diff
@@ -46,14 +47,126 @@ ruleTester.run("no-unneeded-ternary", rule, {
             }]
         },
         {
+            code: "var a = x >= 2 ? true : false;",
+            output: "var a = x >= 2;",
+            errors: [{
+                message: "Unnecessary use of boolean literals in conditional expression.",
+                type: "ConditionalExpression",
+                line: 1,
+                column: 18
+            }]
+        },
+        {
+            code: "var a = x ? true : false;",
+            output: "var a = !!(x);",
+            errors: [{
+                message: "Unnecessary use of boolean literals in conditional expression.",
+                type: "ConditionalExpression",
+                line: 1,
+                column: 13
+            }]
+        },
+        {
+            code: "var a = x === 1 ? false : true;",
+            output: "var a = x !== 1;",
+            errors: [{
+                message: "Unnecessary use of boolean literals in conditional expression.",
+                type: "ConditionalExpression",
+                line: 1,
+                column: 19
+            }]
+        },
+        {
+            code: "var a = x != 1 ? false : true;",
+            output: "var a = x == 1;",
+            errors: [{
+                message: "Unnecessary use of boolean literals in conditional expression.",
+                type: "ConditionalExpression",
+                line: 1,
+                column: 18
+            }]
+        },
+        {
+            code: "var a = foo() ? false : true;",
+            output: "var a = !(foo());",
```

Comment:
I'm a bit concerned that the fix is inserting parentheses that aren't necessary. Again, this seems like more than the rule should be doing for an autofix.